snuba/datasets/discover.py

```diff
@@ -0,0 +1,247 @@
+            ('exception_frames.stack_level', Nullable(UInt(16))),
+            ('modules.name', String()),
+            ('modules.version', String()),
+        ])
+
+    def get_transactions_only_columns(self) -> ColumnSet:
+        return ColumnSet([
+            ('trace_id', Nullable(UUID())),
+            ('span_id', Nullable(UInt(64))),
+            ('transaction_hash', Nullable(UInt(64))),
+            ('transaction_op', Nullable(String())),
+            ('start_ts', Nullable(DateTime())),
+            ('start_ms', Nullable(UInt(16))),
+            ('finish_ts', Nullable(DateTime())),
+            ('finish_ms', Nullable(UInt(16))),
+            ('duration', Nullable(UInt(32))),
+        ])
+
+
+class DiscoverDataset(TimeSeriesDataset):
+    """
+    Experimental dataset for Discover that maps the columns of Events and
+    Transactions into a standard format and sends a query to one of the 2 tables
+    depending on the conditions detected.
+    """
+
+    def __init__(self) -> None:
+        super().__init__(
+            dataset_schemas=DatasetSchemas(
+                read_schema=DiscoverSchema(),
+                write_schema=None,
+            ),
+            time_group_columns={
+                'time': 'timestamp',
+                'bucketed_end': 'finish_ts'
+            },
+            time_parse_columns=['timestamp', 'start_ts', 'finish_ts'],
+        )
+
+    def get_query_processors(self) -> Sequence[QueryProcessor]:
+        return [
+            DiscoverProcessor(),
+        ]
+
+    def get_extensions(self) -> Mapping[str, QueryExtension]:
+        return {
+            'project': ProjectExtension(
+                processor=ProjectWithGroupsProcessor(project_column='project_id')
+            ),
+            'timeseries': TimeSeriesExtension(
+                default_granularity=3600,
+                default_window=timedelta(days=5),
+                timestamp_column='timestamp',
+            ),
+        }
+
+    def column_expr(self, column_name, query: Query, parsing_context: ParsingContext):
+        detected_dataset = detect_dataset(query)
+
+        if detected_dataset == TRANSACTIONS:
+            if column_name == 'type':
+                return "'transaction'"
+            if column_name == 'timestamp':
+                return 'finish_ts'
+            if column_name == 'sentry:release':
+                return 'release'
+            if column_name == 'sentry:dist':
+                return 'dist'
+            if column_name == 'sentry:user':
+                return 'user'
+            if column_name == 'username':
+                return 'user_name'
+            if column_name == 'email':
+                return 'user_email'
+            if column_name == 'transaction':
+                return 'transaction_name'
+            if self.get_dataset_schemas() \
+                    .get_read_schema() \
+                    .get_events_only_columns() \
+                    .get(column_name):
+                return 'NULL'
```

Comment:
Does this query works if I run something like 
select duration, count(*) FROM sentry where type != transaction group by duration

Reply:
Yes it yields a result that looks like `[{"duration": None, "count": 999}]`